libs/react/src/lib/form/formItems.tsx

```diff
@@ -0,0 +1,48 @@
+import { ValidatorRules } from '@sebgroup/extract'
+import React from 'react'
+import { useFormContext } from './formContext'
+import { IValidator } from './types'
+import { validateInputValue } from './validateInput'
+
+export interface FormItemsProps {
+  children: React.ReactNode
+  validate?: IValidator
+  name: string
+}
+
+export const FormItems: React.FC<FormItemsProps> = ({
+  children,
+  validate,
+  name,
+}) => {
+  const { setValues, setErrors, setFields, errors } = useFormContext()
+
+  React.useEffect(() => {
+    setFields((fields: Record<string, any>) => ({
+      ...fields,
+      [name]: validate?.rules,
+    }))
+  }, [])
```

Comment:
given that we are making side effect changes onto context, was wondering if it make sense to undo the side effect when the component is being unmounted? similar to how we remove event listener when component is unmounted.

Reply:
Not too sure about this, should I be returning something here i.e : `return () =>  ....` ?

Reply:
yes, let say if the `FormItems` was dynamically removed, should the `fields` from form context still retain the rules and errors for it?

Reply:
Ahh correct, I should remove it whenever the component unmount